### ISSUE TITLE
How to debug google play services

### ISSUE THREAD

**Issue by ryan-linehan** [mauville-technologies/PGSGP#2]
Hello! You seem to have the most active fork of this google play services repository so I'm going to ask my question here hope that is okay. I am currently struggling to get google play services up and running for the demo application. The readme states that we should be using a OAuth2 client credential that is of type "Web application". Is that correct? It seems a bit odd considering this is an android application and plug-in that we are using.

Also, would this work for a godot 3.3 project?

**Comment by dpnnl** [comment]
Hi! Did you get a solution?

**Comment by ryan-linehan** [comment]
Unfortunately I have not been able to connect to google play services using this repo. I'm currently getting error code 4 even after doing everything mention in the repo owners video. I have also gotten error code 10. Unfortunately google does not provide great error messages detailing why is wrong. I'll let you know if I figure it out

**Comment by dpnnl** [comment]
Thanks for the answer, I tried to redo everything and in the end error 10 always appears, which made me come to the conclusion that the problem is not on my part.
And please let me know if you get something yes 😩

**Comment by dpnnl** [comment]
Hello again.  I managed to solve it, apparently it was a conflict issue between the Play Games ID and the AdMob. Hope this can help.

**Comment by ryan-linehan** [comment]
How did you end up resolving the conflict? Unsure what that means exactly

**Comment by dpnnl** [comment]
The issue was resolved by placing the AdMob ID in the Strings.xml file, looking like this:

`< string name = "app_id" translatable = "false">0000< / string >

< string name = "app_id_admob" translatable = "false">0000< / string >`

**Comment by ryan-linehan** [comment]
I finally figured my own issue out. I am using godot mono (unsure if this contributed to my woes) but when I was running the game on the emulator it seems it was not using the correct keystore to sign it. I am unsure which keystore godot was using to sign it since both my editor settings and where my export settings were pointed were the same. What I did to fix it was exported the project to APK then  used 7zip to unzip the apk file. Next I opened up a command prompt in the unzipped apk location and used this command: keytool -printcert -file META-INF/CERT.RSA (got from: https://developers.google.com/games/services/android/troubleshooting) then saw that the SHA1 value was different than what I was expecting. Copied that SHA1 value into my google play console and bam it worked.
